### PR TITLE
arch/arm: unify compile flags to common/Toolchain.defs 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

arch/arm: unify compile flags to common/Toolchain.defs

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: arch/arm/Kconfig [hunk 1]
```diff
@@ -6,6 +6,44 @@
 if ARCH_ARM
 comment "ARM Options"
 
+choice
+	prompt "Toolchain Selection"
+	default TOOLCHAIN_GNU_EABI
+
+config TOOLCHAIN_IAR
```
Comment (xiaoxiang781216) [inline arch/arm/Kconfig:13]: add ARM_

File: Kconfig [hunk 2]
```diff
@@ -1980,7 +1980,7 @@ config DEBUG_OPTLEVEL
 config DEBUG_OPT_UNUSED_SECTIONS
 	bool "Optimization to eliminate the unused input sections"
 	default y
-	depends on ARCH_TOOLCHAIN_GNU
+	depends on TOOLCHAIN_IS_GNU
```
Comment (xiaoxiang781216) [inline Kconfig:1983]: should we keep ARCH_ prefix

Comment (anchao) [reply Kconfig]: Done

**Comment by xiaoxiang781216** [comment]
https://github.com/apache/incubator-nuttx/pull/7100

**Pull request closed without merging**